airflow/tracer.py

```diff
@@ -0,0 +1,23 @@
+from opentelemetry import trace
+from opentelemetry.instrumentation.flask import FlaskInstrumentor
+from opentelemetry.instrumentation.requests import RequestsInstrumentor
+from opentelemetry.exporter.jaeger.thrift import JaegerExporter
+from opentelemetry.sdk.resources import SERVICE_NAME, Resource
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor
+
+def init(service_name, metric_port):
+    jaeger_exporter = JaegerExporter(
```

Comment:
I'd consider using the `ConsoleSpanExporter` instead of Jaeger. Jaeger is great but requires extra steps like the server in the docker-compose file. The ConsoleSpan on the other hand just prints Open Telemetry spans to STDOUT.

Will email about this too

Reply:
I think the idea of opentelemetry is to provide a vendor-agnostic _agent_ on the localhost, which can forward the data to an external server, which could be either Jaeger or any other collector. 
So you don't need to run the whole Jaeger server (all-in-one) inside airflows own docker-compose, only the agent is required. 
Jaeger ships with such an agent and there is also one provided by opentelemetry, which can forward to Jaeger. Attaching these to console output could be difficult to not get mixed up with logging i assume?

As long as there is a way to configure for forwarding the telemetry to a server running outside of the airflow docker-compose it's good :)

(Sorry for drive-by-comments without knowing the full context and goals here, feel free to ignore)

Reply:
See the mail - I explained some details there. We alreeady have 'integrations" in airflow (separate compose files) and we have a mechanism to enable/disable them in the dev environment - the docker-compose we are talking about is the "Breeze" compose (which is Airflow's development environment and it allows to start whole airflow with optional integrations (kerberos, mongo, different databases etc.) so Jaeger (as server) will eventually have it's own compose file (as all other integrations in Airflow Breeze). And yeah in airflow we will only have (and also as an optional extra) jaeger agent (in OT nomenclature - it is exporter) - and possibly few other exporters.

We even have task for that, to modularise the access. It's not super straightforward though, as we'eve learned that with Gunicorn (and likely Airflow worker model where we use forking), we need to add some extra "exporter-specific" initialization after fork (https://opentelemetry-python.readthedocs.io/en/latest/examples/fork-process-model/README.html) but we will get there eventually.

Reply:
And yeah - another tasks is to integrate Airflow with "services" who handle OT - (Datadog and others) - so totally Airlfow will be able to connect to external OT consumers and Jaeger is really there to make it easy to automatically start development environment.